docs/api/hardhat/plugins.md

```diff
@@ -411,3 +411,98 @@ expect(await contract.getAmount()).to.equal(100);
+- `url` is a field with the URL of the zkSync node in the case of the zkSync network (with `zksync` flag set to `true`), or the URL of the Ethereum node. This field is required for all zkSync and Ethereum networks used by this plugin.
+- `zksync` is a flag to indicate if the network represents zkSync network configuration. This field needs to be set to `true` for all zkSync networks. If you want to run a `hardhat-etherscan` verification, this field needs to be set to `false`. If set to `true`, the verification process will always try to run the verification process on the zkSync network.
+- `verifyURL` is a field that points to the verification endpoint for the specific zkSync network. This parameter is optional, and its default value is the testnet verification url (`https://zksync2-testnet-explorer.zksync.dev/contract_verification`). In a case of a need to use a different url, aside from changing the parameter, a user also has to add `import "./type-extensions.ts` command to the `index.ts` file in the verify plugin.
+
+
+If you want to verify a smart contract on the Ethereum in the same project, it is important to add `etherscan` field in the `hardhat.config.ts` file:
+
+```typescript
+
+networks: {
+    ...
+},
+etherscan: {
+  apiKey: //<Your API key for Etherscan>,
+},
+
+```
+
+### Commands
+
+`hardhat verify --network <network> <contract address>` - verifies the contract on the given network with the given contract's address. </br>
+Note: When run like this, the verify task will try to compare compiled bytecode of all the contracts in your local setup to the deployed bytecode of the contract you are trying to verify. If there is no match, it will report an error.
+
+With the ```--contract``` parameter you can also specify which contract from your local setup you want to verify by specifying its Fully qualified name. Fully qualified name structure looks like this: "contracts/AContract.sol:TheContract" </br>
+
+Example: ``` yarn hardhat verify --network <network> <contract address> --contract <fully qualified  name>```
+
+If your contract was deployed with the specific constructor arguments, you need to specify them when running the verify task. For example: <br/>
+```yarn hardhat verify --network testnet 0x7cf08341524AAF292255F3ecD435f8EE1a910AbF "Hi there!"```
+
+If your constructor takes a complex argument list, you can write a separate javascript module to export it. <br/>
+For example, create an `arguments.js` file with the following structure:
+
+```typescript
+module.exports = [
+  "a string argument",
+  "0xabcdef",
+  "42",
+  {
+    property1: "one",
+    property2: 2
+  }
+];
+
+```
+Include it in the verify function call by adding a new parameter: `--constructor-args arguments.js`:
+```yarn hardhat verify --network testnet 0x7cf08341524AAF292288F3ecD435f8EE1a910AbF --constructor-args arguments.js"```
```

Comment:
I guess we should also include the programmatically verification method here as well

Reply:
Sure, I added it in the last commit.